### PR TITLE
h14

### PR DESCRIPTION
**Pull request by dr-programmer**

### CODE REVIEW

**Comment by dr-programmer** [comment]
@VladislavVulchevChaos

File: SourceCode/h14/optimizations_02.cpp [hunk 1]
```diff
@@ -0,0 +1,1731 @@
...
+    struct AABB& getAABB() {
+        return aabb;
+    }
+    std::vector<struct MeshToTriangle>& getIntersected() {
+        return intersected;
+    }
+};
+
+class KDTree {
+    const int maxDepth = 3;
```
Comment (VladislavVulchevChaos) [inline SourceCode/h14/optimizations_02.cpp:1073]: Seems like a small maximum depth, if you haven't experiment with larger one

File: SourceCode/h14/improvements.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Without opmizations: 662824202 us (microseconds)
+With bucket rendering: 160644341 us (microseconds)
+With bucket rendering and AABB intersection: 69289984 us (microseconds)
+
+With bucket rendering, AABB intersection and KDTree acceleration structure: 
+    13057210 us (microseconds)
```
Comment (VladislavVulchevChaos) [inline SourceCode/h14/improvements.txt:6]: Very nice improvement :)

File: SourceCode/h14/optimizations_02.cpp (hunk 1 above)
Comment (dr-programmer) [reply SourceCode/h14/optimizations_02.cpp:1073]: Yeah, the improvement written in the improvements.txt file is achieved with a depth of 12. @VladislavVulchevChaos

File: SourceCode/h14/improvements.txt (hunk 2 above)
Comment (dr-programmer) [reply SourceCode/h14/improvements.txt:6]: Thanks!

**Pull request merged**